### PR TITLE
[FLINK-25799] [docs] Translate table/filesystem.md page into Chinese.

### PR DESCRIPTION
**Pull request by MrWhiteSike**
## What is the purpose of the change

Translate table/filesystem.md page into Chinese.

## Brief change log

  - file location: flink/docs/content.zh/docs/connectors/table/filesystem.md

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by MrWhiteSike** [comment]
Hi, [@wuchong](https://github.com/wuchong) Would you mind helping to check this pr?

**Comment by MrWhiteSike** [comment]
[@flinkbot](https://github.com/flinkbot)  run Azure

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 1]
```diff
@@ -24,16 +24,15 @@ specific language governing permissions and limitations
...
-supported by the [Flink FileSystem abstraction]({{< ref "docs/deployment/filesystems/overview" >}}).
+# 文件系统 SQL 连接器
 
-The file system connector itself is included in Flink and does not require an additional dependency.
-The corresponding jar can be found in the Flink distribution inside the `/lib` directory.
-A corresponding format needs to be specified for reading and writing rows from and to a file system.
+此连接器提供了对 [Flink FileSystem abstraction]({{< ref "docs/deployment/filesystems/overview" >}}) 支持的文件系统中分区文件的访问。
 
-The file system connector allows for reading and writing from a local or distributed filesystem. A filesystem table can be defined as:
+在 Flink 中包含了该文件系统连接器，不需要添加额外的依赖。相应的 jar 包可以在 Flink 工程项目的 `/lib` 目录下找到。从文件系统中读取或者写入行时，需要指定相应的格式。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:33]:
```suggestion
在 Flink 中包含了该文件系统连接器，不需要添加额外的依赖。相应的 jar 包可以在 Flink 工程项目的 `/lib` 目录下找到。从文件系统中读取或者向文件系统中写入行时，需要指定相应的 format。
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 2]
```diff
@@ -43,37 +42,36 @@ CREATE TABLE MyUserTable (
...
 ```
 
 {{< hint info >}}
-Make sure to include [Flink File System specific dependencies]({{< ref "docs/deployment/filesystems/overview" >}}).
+确定包含 [Flink File System specific dependencies]({{< ref "docs/deployment/filesystems/overview" >}}) 。
 {{< /hint >}}
 
 {{< hint info >}}
-File system sources for streaming is still under development. In the future, the community will add support for common streaming use cases, i.e., partition and directory monitoring.
+基于流的文件系统 sources 仍在开发中。未来，社区将增加对常见的流式用例的支持，例如，对分区和目录的检测等。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:62]: `检测`->`监控`

**Review by RocMarshal — COMMENTED** [review]
@MrWhiteSike Thanks for the review. Left some comments before line 65. PTAL.

**Comment by MrWhiteSike** [comment]
[@RocMarshal](https://github.com/RocMarshal) Thanks for the comments. Please review it again. Thanks

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 3]
```diff
@@ -43,37 +42,36 @@ CREATE TABLE MyUserTable (
...
 
 {{< hint info >}}
-File system sources for streaming is still under development. In the future, the community will add support for common streaming use cases, i.e., partition and directory monitoring.
+基于流的文件系统 sources 仍在开发中。未来，社区将增加对常见的流式用例的支持，例如，对分区和目录的监控等。
 {{< /hint >}}
 
 {{< hint warning >}}
-The behaviour of file system connector is much different from `previous legacy filesystem connector`:
-the path parameter is specified for a directory not for a file and you can't get a human-readable file in the path that you declare.
+文件系统连接器的行为与 `previous legacy filesystem connector` 有很大不同：
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:66]: 文件系统连接器的特性与 `previous legacy filesystem connector` 有很大不同：

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 4]
```diff
@@ -43,37 +42,36 @@ CREATE TABLE MyUserTable (
...
+path 属性指定的是目录，而不是文件，该目录下的文件也不是肉眼可读的。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:67]: Maybe you could translate it in a better way.

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 5]
```diff
@@ -43,37 +42,36 @@ CREATE TABLE MyUserTable (
...
 {{< /hint >}}
 
-## Partition Files
+<a name="partition-files"></a>
 
-Flink's file system partition support uses the standard hive format. However, it does not require partitions to be pre-registered with a table catalog. Partitions are discovered and inferred based on directory structure. For example, a table partitioned based on the directory below would be inferred to contain `datetime` and `hour` partitions.
+## 分区文件
+
+Flink 的文件系统连接器支持分区，使用了标准的 hive 格式。但是，不需要预先注册分区，而是基于目录结构自动做了分区发现。 例如，根据下面的目录结构，进行分区的表将被推断为包含 `datetime` 和 `hour` 分区。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:74]:
```suggestion
Flink 的文件系统连接器支持分区，使用了标准的 hive 格式。但是，不需要预先注册分区到 table catalog，而是基于目录结构自动做了分区发现。 例如，根据下面的目录结构，分区表将被推断包含 `datetime` 和 `hour` 分区。
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 6]
```diff
@@ -88,95 +86,101 @@ path
         ├── part-0.parquet
 ```
 
-The file system table supports both partition inserting and overwrite inserting. See [INSERT Statement]({{< ref "docs/dev/table/sql/insert" >}}). When you insert overwrite to a partitioned table, only the corresponding partition will be overwritten, not the entire table.
+文件系统表支持分区新增插入和分区覆盖插入。请参考 [INSERT Statement]({{< ref "docs/dev/table/sql/insert" >}}) 。当对分区表进行分区覆盖插入时，只有相应的分区会被覆盖，而不是整个表。
+
+<a name="file-formats"></a>
 
-## File Formats
+## 文件格式
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:93]: IMO,  keep the original text and keep ‘Format(s)’ consistent in the whole article.
cc @wuchong @Thesharing

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 7]
```diff
@@ -88,95 +86,101 @@ path
...
-- Canal-JSON: [canal-json]({{< ref "docs/connectors/table/formats/canal" >}}).
-- Raw: [raw]({{< ref "docs/connectors/table/formats/raw" >}}).
+- CSV：[RFC-4180](https://tools.ietf.org/html/rfc4180) 。非压缩格式。
+- JSON：注意，文件系统连接器的 JSON 格式不是传统的标准的 JSON 格式，而是非压缩的。[换行符分割的 JSON](http://jsonlines.org/) 。
+- Avro：[Apache Avro](http://avro.apache.org) 。通过配置 `avro.codec` 属性支持压缩。
+- Parquet：[Apache Parquet](http://parquet.apache.org) 。兼容 hive。
+- Orc：[Apache Orc](http://orc.apache.org) 。兼容 hive。
+- Debezium-JSON：[debezium-json]({{< ref "docs/connectors/table/formats/debezium" >}}) 。
+- Canal-JSON：[canal-json]({{< ref "docs/connectors/table/formats/canal" >}}) 。
+- Raw：[raw]({{< ref "docs/connectors/table/formats/raw" >}}) 。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:104]:
```suggestion
- CSV：[RFC-4180](https://tools.ietf.org/html/rfc4180)。非压缩格式。
- JSON：注意，文件系统连接器的 JSON 格式不是传统的标准的 JSON 格式，而是非压缩的。[换行符分割的 JSON](http://jsonlines.org/)。
- Avro：[Apache Avro](http://avro.apache.org)。通过配置 `avro.codec` 属性支持压缩。
- Parquet：[Apache Parquet](http://parquet.apache.org)。兼容 hive。
- Orc：[Apache Orc](http://orc.apache.org)。兼容 hive。
- Debezium-JSON：[debezium-json]({{< ref "docs/connectors/table/formats/debezium" >}})。
- Canal-JSON：[canal-json]({{< ref "docs/connectors/table/formats/canal" >}})。
- Raw：[raw]({{< ref "docs/connectors/table/formats/raw" >}})。
```

**Review by RocMarshal — COMMENTED** [review]
Thanks for the update.I left some comments before line 108. PTAL.

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 8]
```diff
@@ -88,95 +86,101 @@ path
...
+        如果未设置此属性，只对路径扫描一次，因此将绑定 source。</td>
     </tr>
   </tbody>
 </table>
 
-### Available Metadata
+<a name="available-metadata"></a>
 
-The following connector metadata can be accessed as metadata columns in a table definition. All the metadata are read only.
+### 可提供的 Metadata
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:147]:
```
### 已支持的 Metadata
```

or 

```
### 可用的 Metadata
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 9]
```diff
@@ -88,95 +86,101 @@ path
...
+
+以下连接器 metadata 可以作为表定义对 metadata 进行访问。所有 metadata 都是只读的。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:149]:
```suggestion
以下连接器 metadata 可以在表定义时作为 metadata 列进行访问。所有 metadata 都是只读的。
```

**Review by RocMarshal — COMMENTED** [review]
Thanks for the update. 
Left some comments before line 200. Please let me know what's your opinion.

**Comment by MrWhiteSike** [comment]
[@RocMarshal](https://github.com/RocMarshal) Thanks for the comments. Please review it again.

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 10]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
 ## Streaming Sink
 
-The file system connector supports streaming writes, based on Flink's [FileSystem]({{< ref "docs/connectors/datastream/filesystem" >}}),
-to write records to file. Row-encoded Formats are CSV and JSON. Bulk-encoded Formats are Parquet, ORC and Avro.
+文件系统连接器支持流写入，是基于 Flink 的 [文件系统]({{< ref "docs/connectors/datastream/filesystem" >}}) 写入文件的。CSV 和 JSON 使用的是 Row-encoded Format。Parquet、ORC 和 Avro 使用的是 Bulk-encoded Format。
 
-You can write SQL directly, insert the stream data into the non-partitioned table.
-If it is a partitioned table, you can configure partition related operations. See [Partition Commit](filesystem.html#partition-commit) for details.
+可以直接编写 SQL，将流数据插入到非分区表。
+如果是分区表，可以配置分区操作相关的属性。请参考 [分区提交](#partition-commit) 了解更多详情。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:204]:
```suggestion
如果是分区表，可以配置分区操作相关的属性。请参考[分区提交](#partition-commit)了解更多详情。
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 11]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
-### Rolling Policy
+<a name="rolling-policy"></a>
 
-Data within the partition directories are split into part files. Each partition will contain at least one part file for
-each subtask of the sink that has received data for that partition. The in-progress part file will be closed and additional
-part file will be created according to the configurable rolling policy. The policy rolls part files based on size,
-a timeout that specifies the maximum duration for which a file can be open.
+### 滚动策略
+
+分区目录下的数据被分割到 part 文件中。每个分区对应的 sink 的每个接收到的数据的 subtask 都至少会为该分区生成一个 part 文件。根据可配置的滚动策略，当前 in-progress part 文件将被关闭，生成新的 part 文件。该策略基于大小，和指定的文件可被打开的最大 timeout 时长，来滚动 part 文件。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:210]:
```suggestion
分区目录下的数据被分割到 part 文件中。每个分区对应的 sink 的收到数据的 subtask 都至少会为该分区生成一个 part 文件。根据可配置的滚动策略，当前 in-progress part 文件将被关闭，生成新的 part 文件。该策略基于大小，和指定的文件可被打开的最大 timeout 时长，来滚动 part 文件。
```

Comment (RocMarshal) [reply docs/content.zh/docs/connectors/table/filesystem.md:210]: Before line 337. 
Sorry for the nit part review.  And I'll @ you when the review work was completed. Thanks for the patience.

Comment (MrWhiteSike) [reply docs/content.zh/docs/connectors/table/filesystem.md]: OK !

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 12]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
-There are two types of trigger:
-- The first is partition processing time. It neither requires partition time extraction nor watermark
-  generation. The trigger of partition commit according to partition creation time and current system time. This trigger
-  is more universal, but not so precise. For example, data delay or failover will lead to premature partition commit.
-- The second is the trigger of partition commit according to the time that extracted from partition values and watermark.
-  This requires that your job has watermark generation, and the partition is divided according to time, such as
-  hourly partition or daily partition.
+两种类型分区提交触发器：
+- 第一种是根据分区的处理时间。既不需要额外的分区时间，也不需要 watermark 生成。这种分区提交触发器基于分区创建时间和当前系统时间。
+  这种触发器更具普遍性，但不是很精确。例如，数据延迟或故障将导致过早提交分区。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:338]: `普遍性`->`通用性`

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 13]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
 </table>
 
-There are two types of trigger:
-- The first is partition processing time. It neither requires partition time extraction nor watermark
-  generation. The trigger of partition commit according to partition creation time and current system time. This trigger
-  is more universal, but not so precise. For example, data delay or failover will lead to premature partition commit.
-- The second is the trigger of partition commit according to the time that extracted from partition values and watermark.
-  This requires that your job has watermark generation, and the partition is divided according to time, such as
-  hourly partition or daily partition.
+两种类型分区提交触发器：
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:336]:
```suggestion
Flink 提供了两种类型分区提交触发器：
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 14]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
+- 第一种是根据分区的处理时间。既不需要额外的分区时间，也不需要 watermark 生成。这种分区提交触发器基于分区创建时间和当前系统时间。
+  这种触发器更具普遍性，但不是很精确。例如，数据延迟或故障将导致过早提交分区。
+- 第二种是根据从分区字段提取的时间以及 watermark。
+  这需要 job 支持 watermark 生成，分区是根据时间来切割的，例如，按小时或按天分区。
 
-If you want to let downstream see the partition as soon as possible, no matter whether its data is complete or not:
-- 'sink.partition-commit.trigger'='process-time' (Default value)
-- 'sink.partition-commit.delay'='0s' (Default value)
-  Once there is data in the partition, it will immediately commit. Note: the partition may be committed multiple times.
+如果想让下游尽快感知到分区，不管分区数据是否完整：
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:342]:
```
如果你想让下游尽快感知到分区而并不关心分区数据是否完整：
```
only  a minor comment.

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 15]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
+- 'sink.partition-commit.trigger'='process-time' (默认值)
+- 'sink.partition-commit.delay'='0s' (默认值)
+  一旦数据进入分区，将立即提交分区。注意：这个分区会被提交多次。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:345]:
```suggestion
  一旦数据进入分区，将立即提交分区。注意：这个分区可能会被提交多次。
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 16]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
 
-If you want to let downstream see the partition only when its data is complete, and your job has watermark generation, and you can extract the time from partition values:
+如果想让下游只有在分区数据完整时才感知到分区，并且 job 中有 watermark 生成，也能从分区字段的值中提取到时间：
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:347]:
```suggestion
如果你想让下游只有在分区数据完整时才感知到分区，并且 job 中有 watermark 生成，也能从分区字段的值中提取到时间：
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 17]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
 - 'sink.partition-commit.trigger'='partition-time'
-- 'sink.partition-commit.delay'='1h' ('1h' if your partition is hourly partition, depends on your partition type)
-  This is the most accurate way to commit partition, and it will try to ensure that the committed partitions are as data complete as possible.
+- 'sink.partition-commit.delay'='1h' (根据分区类型指定，如果是按小时分区可配置为 '1h')
+  该方式是最精准地提交分区的方式，尽力确保提交分区的数据尽可能完整。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:350]:
```suggestion
  该方式是最精准地提交分区的方式，尽力确保提交分区的数据完整。
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 18]
```diff
@@ -190,209 +194,217 @@ CREATE TABLE MyUserTableWithFilepath (
...
 
-If you want to let downstream see the partition only when its data is complete, but there is no watermark, or the time cannot be extracted from partition values:
-- 'sink.partition-commit.trigger'='process-time' (Default value)
-- 'sink.partition-commit.delay'='1h' ('1h' if your partition is hourly partition, depends on your partition type)
-  Try to commit partition accurately, but data delay or failover will lead to premature partition commit.
+如果想让下游系统只有在数据完整时才感知到分区，但是没有 watermark，或者无法从分区字段的值中提取时间：
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:352]:
```suggestion
如果你想让下游系统只有在数据完整时才感知到分区，但是没有 watermark，或者无法从分区字段的值中提取时间：
```

**Review by RocMarshal — COMMENTED** [review]
before line 500.

File: docs/content.zh/docs/connectors/table/filesystem.md (hunk 16 above)
Comment (MrWhiteSike) [reply docs/content.zh/docs/connectors/table/filesystem.md:347]: Sorry, I don't think it's necessary to add 'you' in Chinese!

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 19]
```diff
@@ -474,36 +493,40 @@ public class AnalysisCommitPolicy implements PartitionCommitPolicy {
 
 ```
 
+<a name="sink-parallelism"></a>
+
 ## Sink Parallelism
 
-The parallelism of writing files into external file system (including Hive) can be configured by the corresponding table option, which is supported both in streaming mode and in batch mode. By default, the parallelism is configured to being the same as the parallelism of its last upstream chained operator. When the parallelism which is different from the parallelism of the upstream parallelism is configured, the operator of writing files and the operator compacting files (if used) will apply the parallelism.
+在流模式和批模式下，向外部文件系统（包括 hive）写文件时的 parallelism 可以通过相应的 table 属性指定。默认情况下，该 sink parallelism 与上游 chained operator 的 parallelism 一样。当配置了跟上游的 chained operator 不一样的 parallelism 时，写文件和合并文件的算子（如果开启的话）会使用指定的 sink parallelism。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:500]:
```suggestion
在流模式和批模式下，向外部文件系统（包括 hive）写文件时的 parallelism 可以通过相应的 table 配置项指定。默认情况下，该 sink parallelism 与上游 chained operator 的 parallelism 一样。当配置了跟上游的 chained operator 不一样的 parallelism 时，写文件和合并文件的算子（如果开启的话）会使用指定的 sink parallelism。
```
```suggestion
在流模式和批模式下，向外部文件系统（包括 hive）写文件时的 parallelism 可以通过相应的 table 属性指定。默认情况下，该 sink parallelism 与上游 chained operator 的 parallelism 一样。当配置了跟上游的 chained operator 不一样的 parallelism 时，写文件和合并文件的算子（如果开启的话）会使用指定的 sink parallelism。
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 20]
```diff
@@ -474,36 +493,40 @@ public class AnalysisCommitPolicy implements PartitionCommitPolicy {
...
     </tr>
   </thead>
   <tbody>
     <tr>
         <td><h5>sink.parallelism</h5></td>
-        <td style="word-wrap: break-word;">(none)</td>
+        <td style="word-wrap: break-word;">(无)</td>
         <td>Integer</td>
-        <td>Parallelism of writing files into external file system. The value should greater than zero otherwise exception will be thrown.</td>
+        <td> 将文件写入外部文件系统的 parallelism。这个值应该大于0否则抛异常。</td>
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:517]:
```suggestion
        <td>将文件写入外部文件系统的 parallelism。这个值应该大于0否则抛异常。</td>
```
```suggestion
        <td> 将文件写入外部文件系统的 parallelism。这个值应该大于0否则抛异常。</td>
```

File: docs/content.zh/docs/connectors/table/filesystem.md [hunk 21]
```diff
@@ -474,36 +493,40 @@ public class AnalysisCommitPolicy implements PartitionCommitPolicy {
...
-**NOTE:** Currently, Configuring sink parallelism is supported if and only if the changelog mode of upstream is **INSERT-ONLY**. Otherwise, exception will be thrown.
+**注意：** 目前，当且仅当上游的 changelog 模式为 **INSERT-ONLY** 时，才支持配置 sink parallelism。否则，程序将会抛出异常。
+
+<a name="full-example"></a>
 
-## Full Example
+## 完整示例
 
-The below examples show how the file system connector can be used to write a streaming query to write data from Kafka into a file system and runs a batch query to read that data back out.
+以下示例展示了如何使用文件系统连接器编写流式查询语句，将数据从 Kafka 写入文件系统，然后运行批式查询语句将数据读回。
```
Comment (RocMarshal) [inline docs/content.zh/docs/connectors/table/filesystem.md:529]:
```suggestion
以下示例展示了如何使用文件系统连接器编写流式查询语句，将数据从 Kafka 写入文件系统，然后运行批式查询语句读取数据。
```
```suggestion
以下示例展示了如何使用文件系统连接器编写流式查询语句，将数据从 Kafka 写入文件系统，然后运行批式查询语句将数据读回。
```

**Review by RocMarshal — COMMENTED** [review]
@MrWhiteSike FYI, the review from me was completed.
Thanks.

**Comment by MrWhiteSike** [comment]
[@RocMarshal](https://github.com/RocMarshal) Thanks for your comments. Please review it again.

**Comment by RocMarshal** [comment]
IMO,it would be better if you squish the multi commits in thr pr into one commit.

**Comment by MrWhiteSike** [comment]
[@RocMarshal](https://github.com/RocMarshal) Thanks for the comment. I squash some commits to one. IMO, it is the important to focus on the quality of page!  Please let me know what's your opinion.

**Comment by MrWhiteSike** [comment]
[@wuchong](https://github.com/wuchong) Would you mind making the final check on it ? Thanks a lot.

**Review by RocMarshal — APPROVED** [review]
LGTM +1 (non-binding).
@MrWhiteSike Thanks a lot.
It still need a committer to help  make a final check.
Now, ping @wuchong .

**Comment by wuchong** [comment]
Thanks @RocMarshal and @MrWhiteSike ! I will merge it.

**Pull request merged**